### ISSUE TITLE
no-conditional-expect false positive with try-catch

### ISSUE THREAD

**Issue by leeyi45** [vitest-dev/eslint-plugin-vitest#939]
`no-conditional-expect` currently considers the following code as an error:

```ts
try {
  funcThatThrows()
} catch (error) {
  expect(error.message).toEqual('some message')
}
```

Unfortunately if we want to perform multiple assertions using `expect` on the thrown error object, we have to do it in the catch block. I do think that the above situation should be an error, but if the code path "asserts" that an error is thrown then the rule shouldn't error:

```ts
try {
  funcThatThrows()
  expect.fail('Error should have been thrown')
} catch (error) {
  expect(error.message).toEqual('some message')
}
```
After all, the above is equivalent to an `.toThrow` assertion.
